src/components/AssetMatching/components/Toolbox/PrimaryActions.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { FC } from 'react';
+import Styles from './Toolbox.style';
+import { MatchingAction } from '../../types';
+import { useMatchingActions } from '../utils';
+import { generateUniqueID } from '../../../../utils/helpers';
+import { flex } from '../../../../theme/functions';
+
+interface Props {
+  matchingActions: MatchingAction[];
+}
+
+const PrimaryActions: FC<Props> = ({ matchingActions }) => {
+  const primaryActions = matchingActions.slice(0, 2);
+
+  const { actionItems } = useMatchingActions(primaryActions, actionButton => (
+    <div
+      data-testid={'primary_action'}
+      key={generateUniqueID('primary_action')}
```

Comment:
didn't got that part. Why we need a unique key defined here?

Reply:
Checking the `useMatchingActions` hook you will notice that we can provide a callback that enhances each action item with a wrapper element. Therefore we should provide a unique key for each of the list items.